### DISCUSSION TITLE
When will there be a tiddlywiki compiler

### DISCUSSION THREAD

**Discussion by dongrentianyu** [TiddlyWiki/TiddlyWiki5#7410] [Ideas]
The ability of tiddlywiki to program internally is an advantage, but it can also be a dilemma for subsequent advanced users. This is because users who do not know how to program cannot understand what is wrong with the code they have written. Their only options are to continue reading the documentation and other people's examples, or to go to the forums and discuss it.

This is of course fine, but a compiler that reports errors would be even better.

Some users are likely to leave when they don't have the patience to solve code problems. This is why many of my friends can't accept tiddlywiki. tiddlywiki is too complicated and too expensive to learn.

**Comment by Jermolene** [comment]
Hi @dongrentianyu I agree that better debugging tools, including error messages, would improve the experience of learning TiddlyWiki, and would welcome any suggestions.

**Comment by dongrentianyu** [comment]
I think it's important to raise the priority of this task in the beginning, otherwise the more people join into it, the more people will leave. Secondly, are we going to go for a plugin to do this, or are we going to make something like tiddlywiki desktop software to do this. I prefer the second option, to make a software to implement the debug effect. The user puts in some sample entries and then writes a filter or code to determine if the desired effect is achieved.

**Comment by pmario** [comment]
IMO an interesting approach

- New in the v5.3.0 preview: [visible transclusions](https://tiddlywiki.com/prerelease/#Visible%20Transclusions) 
- as discussed at: https://talk.tiddlywiki.org/t/my-favorite-5-3-0-prerelease-feature-x-ray-vision/6811

IMO slightly related issues: 

- [IDEA] TiddlyWiki filter visualizer/helper tool #5058 
- [IDEA] Plugin sourcing, find out UI element is caused by which plugin/tiddler #6345
